src/Turtle/Shell.hs

```diff
@@ -151,6 +154,11 @@ instance MonadManaged Shell where
         x' <- with resource (step x)
         done x' )
 
+#if MIN_VERSION_base(4,9,0)
+instance Fail.MonadFail Shell where
+    fail = fail
```

Comment:
Could you qualify the right-hand side as `Prelude.fail` just so that we don't run the risk of introducing an infinite loop in the future if we import `Control.Monad.Fail.fail` unqualified?